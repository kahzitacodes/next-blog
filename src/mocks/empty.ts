import { EmptyProps } from '@/components/empty'

export const emptyMock: EmptyProps = {
  title: 'Oops!',
  description: 'Page not found'
}
