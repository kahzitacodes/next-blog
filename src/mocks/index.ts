export * from './navigation'
export * from './profile'
export * from './social-links'
export * from './post'
export * from './pre'
export * from './pagination'
