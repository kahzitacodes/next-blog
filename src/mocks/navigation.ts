import { NavItem } from '@/types'

export const navigationMock: NavItem[] = [
  {
    title: 'Home',
    href: '/'
  },
  {
    title: 'About',
    href: '/about'
  },
  {
    title: 'Contact',
    href: '/contact'
  }
]
