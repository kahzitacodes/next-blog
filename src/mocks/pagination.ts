export const paginationMock = {
  currentPage: 1,
  totalPages: 10
}
