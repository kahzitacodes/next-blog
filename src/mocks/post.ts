import { Post } from '@/types'

export const postMock: Post = {
  $frontMatter: {
    date: '2025-04-04',
    description: 'Description',
    image: '/assets/images/image.png',
    title: 'Post title',
    tags: ['react']
  },
  $readingTime: 3,
  body: {
    code: '',
    raw: ''
  },
  id: '1',
  slug: 'post-3'
}
