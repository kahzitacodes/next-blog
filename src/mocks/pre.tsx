import { PreProps } from '@/components/pre'

export const preMock: PreProps = {
  children: <h1>Title</h1>
}
