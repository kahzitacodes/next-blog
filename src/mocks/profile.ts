import type { ProfileProps } from '@/components'

export const profileMock: ProfileProps = {
  avatar: '/assets/images/avatar.jpeg',
  name: 'Karine Pereira',
  description: 'Lorem ipsum dolor sit amet.'
}
