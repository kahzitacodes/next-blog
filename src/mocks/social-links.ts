import { NavItem } from '@/types'

export const socialLinksMock: NavItem[] = [
  {
    title: 'Instagram',
    href: 'http://instagram.com',
    icon: 'Instagram'
  },
  {
    title: 'Twitter',
    href: 'http://twitter.com',
    icon: 'Twitter'
  },
  {
    title: 'LinkedIn',
    href: 'http://linkedin.com',
    icon: 'Linkedin'
  },
  {
    title: 'GitHub',
    href: 'http://github.com',
    icon: 'Github'
  }
]
