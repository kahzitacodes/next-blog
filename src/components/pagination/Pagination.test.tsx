import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import { paginationMock } from '@/mocks'
import { Pagination } from '.'

const mockPush = jest.fn()

jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push: mockPush
  }),
  usePathname: () => '/mock-path'
}))

describe('<Pagination />', () => {
  it('should render successfully', () => {
    const { container } = render(
      <Pagination dataTestId="pagination" {...paginationMock} />
    )

    expect(screen.getByTestId('pagination')).toBeInTheDocument()
    expect(screen.getByText(/1 de 10/i)).toBeInTheDocument()
    expect(screen.getByRole('button', { name: /previous/i })).toBeDisabled()
    expect(screen.getByRole('button', { name: /next/i })).toBeEnabled()

    expect(container).toMatchSnapshot()
  })

  it('should call router.push with next page when "next" is clicked', () => {
    render(<Pagination {...paginationMock} />)

    fireEvent.click(screen.getByRole('button', { name: /next/i }))

    expect(mockPush).toHaveBeenCalledWith('/mock-path?page=2')
    waitFor(() => {
      expect(screen.getByText(/2 de 10/i)).toBeInTheDocument()
    })
  })

  it('"next" should be disabled when current page is the last page', () => {
    render(<Pagination currentPage={10} totalPages={10} />)

    expect(screen.getByRole('button', { name: /previous/i })).toBeEnabled()
    expect(screen.getByRole('button', { name: /next/i })).toBeDisabled()
    expect(screen.getByText(/10 de 10/i)).toBeInTheDocument()
  })
})
