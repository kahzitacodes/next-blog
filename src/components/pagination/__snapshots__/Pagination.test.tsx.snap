// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render successfully 1`] = `
<div>
  <div
    class="flex w-full items-center justify-between py-7"
    data-testid="pagination"
  >
    <button
      class="flex items-center gap-1 text-white font-medium h-[40px] rounded-md w-fit transition-colors duration-200 group disabled:text-zinc-600 disabled:cursor-default disabled:border-zinc-600 bg-transparent px-0 hover:text-link disabled:pointer-events-none group"
      disabled=""
      variant="ghost"
    >
      <div
        class="duration-200' shrink-0 transition-transform duration-200' group-hover:-translate-x-0.5"
      >
        <svg
          class="lucide lucide-arrow-left"
          data-testid="ArrowLeft"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m12 19-7-7 7-7"
          />
          <path
            d="M19 12H5"
          />
        </svg>
      </div>
      Previous
    </button>
    <span
      class=""
    >
      1
       de 
      10
    </span>
    <button
      class="flex items-center gap-1 text-white font-medium h-[40px] rounded-md w-fit transition-colors duration-200 group disabled:text-zinc-600 disabled:cursor-default disabled:border-zinc-600 bg-transparent px-0 hover:text-link disabled:pointer-events-none group"
      variant="ghost"
    >
      Next
      <div
        class="duration-200' shrink-0 transition-transform duration-200' group-hover:translate-x-0.5"
      >
        <svg
          class="lucide lucide-arrow-right"
          data-testid="ArrowRight"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="m12 5 7 7-7 7"
          />
        </svg>
      </div>
    </button>
  </div>
</div>
`;
